### PR TITLE
fix(decorator): implement YAML-safe quoting for string values in decorators

### LINKED ISSUE #1202: Term Decorator Value Corruption and Type Coercion in YAML Round-Trip
The `@Term` decorator experiences widespread value corruption during vocabulary YAML serialization/deserialization, affecting 65% of test cases with type coercion, truncation, and parse errors.

## Expected Behavior

When a Concerto model uses `@Term` decorators with `DecoratorString`, `DecoratorNumber`, `DecoratorBoolean`, or `DecoratorTypeReference` values:
- String values should remain strings regardless of content
- Numeric strings (e.g., `"123"`, `"3.14"`) should not be coerced to numbers
- Boolean-like strings (e.g., `"true"`, `"false"`) should not be coerced to booleans
- Special characters (`:`, `#`, `!`, `[]`, `{}`, `>`) should not cause parse errors or truncation
- Leading/trailing whitespace should be preserved
- Empty strings should remain empty strings, not become `null`
- Zero values (`0`, `"0"`) should be preserved, not become `null`

## Current Behavior

The Term decorator values are corrupted in multiple ways during YAML round-trip:

### 1. Type Coercion Issues
- String `"123"` → Number `123`
- String `"3.14"` → Number `3.14`
- String `"true"` → Boolean `true`
- String `"42"` → Number `42`
- String `"1e10"` → Number `10000000000`
- String `"0x1A"` → Number `26` (hex parsing)
- String `".5"` → Number `0.5`

### 2. Null Coercion Issues
- String `""` (empty) → `null`
- String `"0"` → `null`
- String `"false"` → `null`
- String `"null"` → `null`
- Number `0` → `null`
- Boolean `false` → `null`

### 3. YAML Parse Errors
- Values with colons: `"key: value"` → Error: "Nested mappings not allowed in compact mappings"
- Values with brackets: `"[array] value"` → Error: "Unexpected scalar at node end"
- Values with braces: `"{object} value"` → Error: "Unexpected scalar at node end"
- Values with greater-than: `"> folded text"` → Error: "Not a YAML token"
- Multi-line values → Error: "Implicit keys need to be on single line"

### 4. String Truncation
- `"note # important"` → `"note"` (truncated at `#`)
- `"!important tag"` → `"tag"` (truncated at `!`)
- Complex strings → Severe truncation

### 5. Whitespace Stripping
- `"  leading spaces"` → `"leading spaces"`
- `"trailing spaces  "` → `"trailing spaces"`

## Possible Solution

The root cause appears to be improper YAML serialization of decorator values. Solutions include:

1. **Force String Quoting**: Always emit quoted strings in YAML for `DecoratorString` values
2. **Explicit Type Preservation**: Use YAML tags (e.g., `!!str`) to preserve string types
3. **Escape Special Characters**: Properly escape YAML special characters in decorator values
4. **Custom Serializer**: Implement a custom YAML serializer for decorator arguments that preserves exact types and values

### Minimal Reproduction Example

```javascript
const { ModelManager } = require('@accordproject/concerto-core');

const ctoModel = `namespace test
concept Example {
  @Term("123")  // String that looks like a number
  o String field1
  
  @Term(0)  // Number zero
  o String field2
}`;

const mm = new ModelManager();
mm.addCTOModel(ctoModel, "test.cto");

// Round-trip through vocabulary YAML
const vocab = mm.getModelFile('test').decorators;
// Expected: "123" remains string, 0 remains 0
// Actual: "123" becomes number 123, 0 becomes null
```

## Context (Environment)

This issue affects any Concerto model using `@Term` decorators with:
- String values that resemble other types (numbers, booleans, null)
- Special characters used in YAML syntax
- Empty strings or zero values
- Values requiring preserved whitespace

### Desktop
- OS: macOS
- Node.js: [version used in testing]
- Concerto Core: @accordproject/concerto-core (version from package.json)

### Impact
- **Data Loss**: Values are truncated or lost entirely (e.g., `0` → `null`)
- **Type Safety**: String types are silently converted to numbers/booleans
- **Usability**: Common patterns like numeric IDs as strings fail
- **Reliability**: 64.7% test failure rate indicates systematic issue

## Detailed Description

The bug manifests in the vocabulary management system where `@Term` decorator values undergo serialization to YAML format and subsequent deserialization. The current implementation fails to properly handle:

### Test Coverage
- **Property-level decorators**: 51 test cases
  - ✅ Passed: 18 (35.3%)
  - ❌ Failed: 33 (64.7%)
    - YAML Parse Errors: 7
    - Round-trip Mismatches: 26

- **Declaration-level decorators**: 4 test cases
  - ✅ Passed: 1 (25%)
  - ❌ Failed: 3 (75%)
    - YAML Parse Errors: 2
    - Round-trip Mismatches: 1

### All Decorator Types Affected
1. `DecoratorString`: Type coercion, truncation, parse errors
2. `DecoratorNumber`: Zero becomes null
3. `DecoratorBoolean`: False becomes null
4. `DecoratorTypeReference`: Working correctly ✅

### Critical Failures
The most severe issues are:
1. **Zero values** (both string `"0"` and number `0`) become `null`
2. **Boolean false** becomes `null`
3. **Empty string** becomes `null`
4. **Any numeric-looking string** is coerced to a number
5. **YAML special characters** cause complete parse failures

### Option 1: Force Quoted Scalars (going with this solution)
Modify the YAML serializer to always emit quoted strings for DecoratorString values:
```yaml
# Current (broken)
propertyName: 123

# Fixed
propertyName: "123"
```

### Option 2: Use YAML Type Tags
Explicitly tag decorator values with their intended types:
```yaml
propertyName: !!str "123"
zeroValue: !!int 0
falseValue: !!bool false
```

### Option 3: Custom Serialization Format(pointing towards long terms vocabulary schema)
Use a different serialization format that preserves exact types (e.g., JSON) or implement a custom YAML schema with strict type preservation.

## Steps to Reproduce

1. Run the property tests: `node properties.test.js`
2. Run the declaration tests: `node declaration.test.js`

`properties.test.js`
```js
const { ModelManager, DecoratorManager } = require('@accordproject/concerto-core');
const { VocabularyManager } = require('@accordproject/concerto-vocabulary');

// config
const DECORATOR_MANAGER_OPTIONS = { removeDecoratorsFromModel: true, locale: 'en' };

// cto model template
const NAMESPACE = 'test.properties@1.0.0';
const CONCEPT_NAME = 'Person';
const PROPERTY_NAME = 'name';
const LOCALE = 'en';

// test cases: most are ai generated variations on edge cases for string handling in YAML, plus some specific cases for DecoratorString/Number/Boolean and TypeReference
const propertiesCases = [
 // ─── YAML-SPECIAL CHARACTERS ───
 {
 title: 'Value contains colon',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("key: value")
 o String name
 }`,
 expectedValue: 'key: value'
 },
 {
 title: 'Value contains hash character',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("note # important")
 o String name
 }`,
 expectedValue: 'note # important'
 },
 {
 title: 'Value contains single quotes',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("it's a test")
 o String name
 }`,
 expectedValue: "it's a test"
 },
 {
 title: 'Value contains double quotes',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("say \\"hello\\"")
 o String name
 }`,
 expectedValue: 'say "hello"'
 },
 {
 title: 'Value contains newline characters',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("line1\\nline2\\nline3")
 o String name
 }`,
 expectedValue: 'line1\nline2\nline3'
 },
 {
 title: 'Value contains square brackets',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("[array] value")
 o String name
 }`,
 expectedValue: '[array] value'
 },
 {
 title: 'Value contains curly braces',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("{object} value")
 o String name
 }`,
 expectedValue: '{object} value'
 },
 {
 title: 'Value contains exclamation mark',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("!important tag")
 o String name
 }`,
 expectedValue: '!important tag'
 },
 {
 title: 'Value contains ampersand',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("a & b")
 o String name
 }`,
 expectedValue: 'a & b'
 },
 {
 title: 'Value contains asterisk',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("a * b")
 o String name
 }`,
 expectedValue: 'a * b'
 },
 {
 title: 'Value contains pipe character',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("option1 | option2")
 o String name
 }`,
 expectedValue: 'option1 | option2'
 },
 {
 title: 'Value contains greater-than character',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("> folded text")
 o String name
 }`,
 expectedValue: '> folded text'
 },
 {
 title: 'Value has leading spaces',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term(" leading spaces")
 o String name
 }`,
 expectedValue: ' leading spaces'
 },
 {
 title: 'Value has trailing spaces',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("trailing spaces ")
 o String name
 }`,
 expectedValue: 'trailing spaces '
 },
 {
 title: 'Value contains backslashes',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("path\\\\to\\\\file")
 o String name
 }`,
 expectedValue: 'path\\to\\file'
 },
 {
 title: 'Value contains commas',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("one, two, three")
 o String name
 }`,
 expectedValue: 'one, two, three'
 },
 {
 title: 'Value contains question mark',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("is this valid?")
 o String name
 }`,
 expectedValue: 'is this valid?'
 },

 // ─── YAML TYPE COERCION ───
 {
 title: 'Value looks like YAML boolean "true"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("true")
 o String name
 }`,
 expectedValue: 'true'
 },
 {
 title: 'Value looks like YAML boolean "false"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("false")
 o String name
 }`,
 expectedValue: 'false'
 },
 {
 title: 'Value looks like YAML boolean "yes"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("yes")
 o String name
 }`,
 expectedValue: 'yes'
 },
 {
 title: 'Value looks like YAML boolean "no"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("no")
 o String name
 }`,
 expectedValue: 'no'
 },
 {
 title: 'Value looks like YAML integer "123"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("123")
 o String name
 }`,
 expectedValue: '123'
 },
 {
 title: 'Value looks like YAML float "1.5"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("1.5")
 o String name
 }`,
 expectedValue: '1.5'
 },
 {
 title: 'Value looks like YAML float "3.14"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("3.14")
 o String name
 }`,
 expectedValue: '3.14'
 },

 // ─── EDGE CASES ───
 {
 title: 'Value is empty string',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("")
 o String name
 }`,
 expectedValue: ''
 },
 {
 title: 'Plain string without special characters (control)',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("Simple Label")
 o String name
 }`,
 expectedValue: 'Simple Label'
 },
 {
 title: 'Value is stringified array',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("['a', 'b']")
 o String name
 }`,
 expectedValue: "['a', 'b']"
 },
 {
 title: 'Value contains multi-line YAML-like content',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("Custom Field: Name\\n fieldType: Text\\n properties:\\n - value: Custom Field: Name")
 o String name
 }`,
 expectedValue: 'Custom Field: Name\n fieldType: Text\n properties:\n - value: Custom Field: Name'
 },
 {
 title: 'Value contains ALL special characters combined',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("k:v #c 'q' [a] {b} |p >f &r *s !i ,d ?q")
 o String name
 }`,
 expectedValue: "k:v #c 'q' [a] {b} |p >f &r *s !i ,d ?q"
 },

 // ─── DECORATOR STRING — NUMERIC/BOOLEAN/NULL LOOK-ALIKES ───
 {
 title: 'DecoratorString: string that looks like integer "42"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("42")
 o String name
 }`,
 expectedValue: '42'
 },
 {
 title: 'DecoratorString: string that looks like float "3.14"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("3.14")
 o String name
 }`,
 expectedValue: '3.14'
 },
 {
 title: 'DecoratorString: string that looks like boolean "true"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("true")
 o String name
 }`,
 expectedValue: 'true'
 },
 {
 title: 'DecoratorString: string that looks like boolean "false"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("false")
 o String name
 }`,
 expectedValue: 'false'
 },
 {
 title: 'DecoratorString: plain string (control)',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("Hello World")
 o String name
 }`,
 expectedValue: 'Hello World'
 },
 {
 title: 'DecoratorString: string that looks like YAML null',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("null")
 o String name
 }`,
 expectedValue: 'null'
 },
 {
 title: 'DecoratorString: string zero "0"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("0")
 o String name
 }`,
 expectedValue: '0'
 },
 {
 title: 'DecoratorString: string "Infinity"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("Infinity")
 o String name
 }`,
 expectedValue: 'Infinity'
 },
 {
 title: 'DecoratorString: string "NaN"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("NaN")
 o String name
 }`,
 expectedValue: 'NaN'
 },
 {
 title: 'DecoratorString: leading dot float ".5"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term(".5")
 o String name
 }`,
 expectedValue: '.5'
 },
 {
 title: 'DecoratorString: scientific notation "1e10"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("1e10")
 o String name
 }`,
 expectedValue: '1e10'
 },
 {
 title: 'DecoratorString: negative number "-1"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("-1")
 o String name
 }`,
 expectedValue: '-1'
 },
 {
 title: 'DecoratorString: hex notation "0x1A"',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term("0x1A")
 o String name
 }`,
 expectedValue: '0x1A'
 },

 // ─── DECORATOR NUMBER (non-string argument) ───
 {
 title: 'DecoratorNumber: @Term(42) — integer argument',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term(42)
 o String name
 }`,
 expectedValue: 42
 },
 {
 title: 'DecoratorNumber: @Term(3.14) — float argument',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term(3.14)
 o String name
 }`,
 expectedValue: 3.14
 },
 {
 title: 'DecoratorNumber: @Term(0) — zero',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term(0)
 o String name
 }`,
 expectedValue: 0
 },
 {
 title: 'DecoratorNumber: @Term(-1) — negative number',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term(-1)
 o String name
 }`,
 expectedValue: -1
 },
 {
 title: 'DecoratorNumber: @Term(100.0) — float with trailing zero',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term(100.0)
 o String name
 }`,
 expectedValue: 100.0
 },

 // ─── DECORATOR BOOLEAN (non-string argument) ───
 {
 title: 'DecoratorBoolean: @Term(true) — boolean true',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term(true)
 o String name
 }`,
 expectedValue: true
 },
 {
 title: 'DecoratorBoolean: @Term(false) — boolean false',
 ctoModel: `namespace test.properties@1.0.0
 concept Person {
 @Term(false)
 o String name
 }`,
 expectedValue: false
 },

 // ─── DECORATOR TYPE REFERENCE (no .value field) ─── // TODO: HERE I need a bit of 
 {
 title: 'DecoratorTypeReference: @Term(Name) — type reference',
 ctoModel: `namespace test.properties@1.0.0

 concept Name {
 o String fName
 o String lName
 }

 concept Person {
 @Term(Name)
 o String name
 }`,
 expectedValue: null
 },
 {
 title: 'DecoratorTypeReference: @Term(Name[]) — array type reference',
 ctoModel: `namespace test.properties@1.0.0

 concept Name {
 o String fName
 o String lName
 } 

 concept Person {
 @Term(Name[])
 o String name
 }`,
 expectedValue: null // TODO: understand why is null expected here, I would have thought it should be the string "Name[]"
 },
];


function testProperties(testTitle, ctoModel, expectedValue) {
 console.log(`=== 🧪 TC : ${testTitle} ===`);

 // load the model
 const mm = new ModelManager();
 mm.addCTOModel(ctoModel, `test-properties.cto`);

 // extract vocabularies using DecoratorManager
 let decoratorManagerResult;
 try {
 decoratorManagerResult = DecoratorManager.extractVocabularies(mm, DECORATOR_MANAGER_OPTIONS);
 } catch (e) {
 console.log('Test failed ❌');
 console.log('☢️ DM ERROR:');
 console.log(e.message);
 return;
 }

 const extractedVocab = decoratorManagerResult.vocabularies[0];

 // test round trip through VocabularyManager
 let vmResult;
 try {
 const vm = new VocabularyManager();
 vm.addVocabulary(extractedVocab);
 vmResult = vm.getTerm(NAMESPACE, LOCALE, CONCEPT_NAME, PROPERTY_NAME);
 } catch (e) {
 console.log('Test failed ❌');
 console.log('☢️ VM ERROR:');
 console.log(e.message);
 console.log('=== ERROR EXTRACTED VOCAB ===');
 console.log(extractedVocab);
 console.log('=== === ===');
 return;
 }


 // asserting
 const roundTripBug = vmResult !== expectedValue;
 // TODO: check if extracted vocab is yaml

 // log results
 if (!roundTripBug) {
 console.log('Test passed ✅');
 return;
 } else {
 console.log('Test failed ❌');
 }

 console.log(' → VocabularyManager round-trip mismatch');
 console.log('VocabMgr result :', JSON.stringify(vmResult));
 console.log('Expected :', JSON.stringify(expectedValue));
}

function main() {
 propertiesCases.forEach(c => testProperties(c.title, c.ctoModel, c.expectedValue));
}

main();
```

`declaration.test.js`
```js
const { ModelManager, DecoratorManager } = require('@accordproject/concerto-core');
const { VocabularyManager } = require('@accordproject/concerto-vocabulary');

// config
const DECORATOR_MANAGER_OPTIONS = { removeDecoratorsFromModel: true, locale: 'en' };

// cto model template
const NAMESPACE = 'test.declaration@1.0.0';
const CONCEPT_NAME = 'Person';
const PROPERTY_NAME = 'name';
const LOCALE = 'en';

// test cases: most are ai generated variations on edge cases for string handling in YAML, plus some specific cases for DecoratorString/Number/Boolean and TypeReference
const declarationCases = [
 {
 title: 'Declaration @Term plain (control)',
 ctoModel: `namespace test.declaration@1.0.0
 @Term("My Declaration")
 concept Person {
 o String name
 }`,
 expectedValue: 'My Declaration'
 },
 {
 title: 'Declaration @Term with colon',
 ctoModel: `namespace test.declaration@1.0.0
 @Term("My Concept: The First")
 concept Person {
 o String name
 }`,
 expectedValue: 'My Concept: The First'
 },
 {
 title: 'Declaration @Term with hash',
 ctoModel: `namespace test.declaration@1.0.0
 @Term("Section # 1")
 concept Person {
 o String name
 }`,
 expectedValue: 'Section # 1'
 },
 {
 title: 'Declaration @Term with brackets',
 ctoModel: `namespace test.declaration@1.0.0
 @Term("[Category] Items")
 concept Person {
 o String name
 }`,
 expectedValue: '[Category] Items'
 }
];

function testDeclarations(testTitle, ctoModel, expectedValue) {
 console.log(`=== 🧪 TC : ${testTitle} ===`);

 // load the model
 const mm = new ModelManager();
 mm.addCTOModel(ctoModel, `test-declarations.cto`);

 // extract vocabularies using DecoratorManager
 let decoratorManagerResult;
 try {
 decoratorManagerResult = DecoratorManager.extractVocabularies(mm, DECORATOR_MANAGER_OPTIONS);
 } catch (e) {
 console.log('Test failed ❌');
 console.log('☢️ DM ERROR:');
 console.log(e.message);
 return;
 }

 const extractedVocab = decoratorManagerResult.vocabularies[0];

 // test round trip through VocabularyManager
 let vmResult;
 try {
 const vm = new VocabularyManager();
 vm.addVocabulary(extractedVocab);
 vmResult = vm.getTerm(NAMESPACE, LOCALE, CONCEPT_NAME);
 } catch (e) {
 console.log('Test failed ❌');
 console.log('☢️ VM ERROR:');
 console.log(e.message);
 console.log('=== ERROR EXTRACTED VOCAB ===');
 console.log(extractedVocab);
 console.log('=== === ===');
 return;
 }


 // asserting
 const roundTripBug = vmResult !== expectedValue;
 // TODO: check if extracted vocab is yaml

 // log results
 if (!roundTripBug) {
 console.log('Test passed ✅');
 return;
 } else {
 console.log('Test failed ❌');
 }

 console.log(' → VocabularyManager round-trip mismatch');
 console.log('VocabMgr result :', JSON.stringify(vmResult));
 console.log('Expected :', JSON.stringify(expectedValue));
}

function main() {
 declarationCases.forEach(c => testDeclarations(c.title, c.ctoModel, c.expectedValue));
}

main();
```

### PR DESCRIPTION
**Pull request by mon7792**
# Closes #1202

## What was the problem?

In Concerto, you can annotate model fields with `@Term` decorators to give them human-readable labels — for example:

```
@Term("Custom Field: Name")
o String fieldName
```

When `DecoratorExtractor` converts these annotations into a YAML vocabulary file, it was writing the value as-is:

```yaml
fieldName: Custom Field: Name   # ❌ broken — YAML sees two keys here
```

YAML treats `:` followed by a space as a key-value separator, so this produces invalid YAML. The same problem existed for many other special characters — quotes, booleans like `true`/`false`, numbers like `1.5`, empty strings, and more.

A second bug: if a term value was falsy (empty string `""`, number `0`, boolean `false`), the code treated it as "not set" and skipped it or fell back to a parent locale — losing the value silently.

---

### 1. YAML-safe quoting

Added a `quoteStringValue()` function that checks whether a string needs quoting before writing it to YAML. If it does, the value is wrapped in double quotes with special characters properly escaped:

```yaml
fieldName: "Custom Field: Name"   # ✅ valid YAML
```

It handles all the YAML hazard categories:
- Inline special chars: `:`, `#`, `'`, `"`, newlines, tabs, carriage returns, backslashes
- First-character triggers: `[`, `!`, `&`, `*`, `%`, `>`, `|`, etc.
- Reserved words: `true`, `false`, `yes`, `no`, `null`, `~` (YAML 1.1 parses these as non-string)
- Numeric-looking strings: `1.5`, `-1`, `1e10`, `.5`
- Empty string (bare empty value becomes `null` in YAML without quotes)

Numbers and booleans from `Term_*` decorators (e.g. `@Term_priority(1)`) are left unquoted — they are already valid YAML scalars.

### 2. Falsy value handling

Fixed three places where JavaScript's truthiness check was incorrectly treating `""`, `0`, and `false` as "missing":

- `vocabulary.ts`: changed `d[propertyName]` → `propertyName in d` to detect key existence instead of value truthiness (applied to both `getTerm` and `getElementTerms`)
- `vocabularymanager.ts`: changed `if (term)` → `if (term != null)` in both `getTerm` and `getTerms` so falsy terms are returned instead of falling back to the parent locale
- Return types of `getTerm` and `resolveTerm` broadened from `string | null` to `string | number | boolean | null`

---

### Decorator extractor (YAML quoting)
Five new fixture `.cto` files exercise every edge case. Tests parse the YAML output with a real YAML parser and assert the original values round-trip intact — not just that the output contains certain strings.

### Vocabulary falsy value handling (15 new tests)
**`vocabulary.js`** — `Vocabulary.getTerm` / `Vocabulary.getElementTerms`:
- Property with `false`, `0`, `""` as term value — verifies the `propertyName in d` fix finds keys with falsy values
- Declaration with `false` as term value
- `getElementTerms` with `false`, `0`, `""` — same fix path

**`vocabularymanager.js`** — `VocabularyManager.getTerm` / `VocabularyManager.getTerms`:
- Each test sets a falsy value (`false`, `0`, `""`) in `en-gb` vocab and a non-falsy fallback in the parent `en` vocab
- Asserts the falsy value is returned as-is, confirming no incorrect locale fallback occurs

### Flags for reviewers
- YAML 1.1 reserved words (`true`, `false`, `yes`, `no`, `null`, etc.) are now **quoted** in output — any downstream parser that relied on them being coerced to booleans/null will see strings instead

### Related Issues
- Issue #1183
- Pull Request #1184

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:monish/term-decor-fix`

### CODE REVIEW

File: packages/concerto-core/src/decoratorextractor.ts [hunk 1]
```diff
@@ -133,27 +177,27 @@ class DecoratorExtractor {
...
                 if (vocabObject.namespace.term){
-                    strVoc += `term: ${vocabObject.namespace.term}\n`;
+                    strVoc += `term: ${this.quoteStringValue(vocabObject.namespace.term, vocabObject.namespace.term_type)}\n`;
                 }
-                let otherProps = Object.keys(vocabObject.namespace).filter((str)=>str !== 'term');
+                let otherProps = Object.keys(vocabObject.namespace).filter((str)=>str !== 'term' && !str.endsWith('_type'));
                 otherProps.forEach(key =>{
-                    strVoc += `${key}: ${vocabObject.namespace[key]}\n`;
+                    strVoc += `${key}: ${this.quoteStringValue(vocabObject.namespace[key], vocabObject.namespace[`${key}_type`])}\n`;
                 });
```
[the remark this answers is not in the document]
Comment (mon7792) [reply packages/concerto-core/src/decoratorextractor.ts:163]: no longer valid. moved away from this solution altogether. unnecessary for current implementation.

File: packages/concerto-core/src/decoratorextractor.ts [hunk 2]
```diff
@@ -162,10 +206,10 @@ class DecoratorExtractor {
                         }
                         strVoc += '    properties:\n';
                         Object.keys(vocabObject.declarations[decl].propertyVocabs).forEach(prop =>{
-                            strVoc += `      - ${prop}: ${vocabObject.declarations[decl].propertyVocabs[prop].term || ''}\n`;
-                            const otherProps = Object.keys(vocabObject.declarations[decl].propertyVocabs[prop]).filter((str)=>str !== 'term');
+                            strVoc += `      - ${prop}: ${this.quoteStringValue(vocabObject.declarations[decl].propertyVocabs[prop].term, vocabObject.declarations[decl].propertyVocabs[prop].term_type) || ''}\n`;
```
[the remark this answers is not in the document]
Comment (mon7792) [reply packages/concerto-core/src/decoratorextractor.ts]: rectified.

File: packages/concerto-vocabulary/src/vocabulary.ts [hunk 3]
```diff
@@ -131,7 +131,7 @@ class Vocabulary {
             return identifier ? decl[identifier] : decl[declarationName];
         }
         else {
-            const property = decl.properties ? decl.properties.find((d: any) => d[propertyName]) : null;
+            const property = decl.properties ? decl.properties.find((d: any) => propertyName in d) : null;
             return property ? identifier ? property[identifier] : property[propertyName] : null;
```
[the remark this answers is not in the document]
Comment (mon7792) [reply packages/concerto-vocabulary/src/vocabulary.ts:135]: done

File: packages/concerto-core/src/decoratorextractor.ts [hunk 4]
```diff
@@ -133,27 +177,27 @@ class DecoratorExtractor {
...
+                    strVoc += `${key}: ${this.quoteStringValue(vocabObject.namespace[key], vocabObject.namespace[`${key}_type`])}\n`;
                 });
             }
             if (vocabObject.declarations && Object.keys(vocabObject.declarations).length > 0 ){
                 strVoc = strVoc + 'declarations:\n';
                 Object.keys(vocabObject.declarations).forEach(decl =>{
                     if (vocabObject.declarations[decl].term){
-                        strVoc += `  - ${decl}: ${vocabObject.declarations[decl].term}\n`;
+                        strVoc += `  - ${decl}: ${this.quoteStringValue(vocabObject.declarations[decl].term, vocabObject.declarations[decl].term_type)}\n`;
                     }
```
[the remark this answers is not in the document]
Comment (mon7792) [reply packages/concerto-core/src/decoratorextractor.ts:170]: done

File: packages/concerto-core/src/decoratorextractor.ts [hunk 5]
```diff
@@ -70,6 +80,40 @@ class DecoratorExtractor {
...
+
+        if (!needsQuote) return str;
+
+        const escaped = str
+            .replace(/\\/g, '\\\\')
+            .replace(/"/g, '\\"')
+            .replace(/\n/g, '\\n')
+            .replace(/\r/g, '\\r')
+            .replace(/\t/g, '\\t');
+        return `"${escaped}"`;
```
Comment (mttrbrts) [inline packages/concerto-core/src/decoratorextractor.ts]: Can we replace this with `YAML.stringify`? I worry about the maintenance, and edge cases that we've missed

Comment (mon7792) [reply packages/concerto-core/src/decoratorextractor.ts]: This makes sense.
made the changes. 

following cases are tested as part of workflow.

| Original | Embedded as | Reason |
|---|---|---|
| *(empty string)* | `""` | Empty scalar → YAML null without quotes |
| `·leading and trailing·` | `" leading and trailing "` | Leading/trailing whitespace |
| `- foo` | `"- foo"` | Block sequence indicator |
| `---` | `"---"` | Document start marker |
| `...` | `"..."` | Document end marker |
| `-1.5` | `"-1.5"` | Numeric (negative float) |
| `.5` | `".5"` | Numeric (float, no leading zero) |
| `1e10` | `"1e10"` | Numeric (scientific notation) |
| `0x1A` | `"0x1A"` | Numeric (hex) |
| `0o10` | `"0o10"` | Numeric (octal) |
| `~` | `"~"` | YAML null shorthand |
| `NULL` | `"NULL"` | YAML null keyword |
| `TRUE` | `"TRUE"` | YAML boolean keyword |
| `[unclosed bracket` | `"[unclosed bracket"` | Flow sequence indicator |
| `[primary, secondary]` | `"[primary, secondary]"` | Flow sequence |
| `{key: val` | `"{key: val"` | Flow mapping indicator |
| `foo:` | `"foo:"` | Trailing colon (mapping key) |
| `Field: Label` | `"Field: Label"` | Colon-space (mapping key) |
| `Key: Identifier` | `"Key: Identifier"` | Colon-space (mapping key) |
| `My Namespace: Title` | `"My Namespace: Title"` | Colon-space (mapping key) |
| `Product: Details` | `"Product: Details"` | Colon-space (mapping key) |
| `Value: Data` | `"Value: Data"` | Colon-space (mapping key) |
| `status: active # reviewed` | `"status: active # reviewed"` | Colon-space + comment |
| `> folded block` | `"> folded block"` | Block folded scalar indicator |
| `\| literal block` | `"\| literal block"` | Block literal scalar indicator |
| `!tag handle` | `"!tag handle"` | Tag indicator |
| `*alias ref` | `"*alias ref"` | Alias indicator |
| `&anchor ref` | `"&anchor ref"` | Anchor indicator |
| `%directive` | `"%directive"` | Directive indicator |
| `? key` | `"? key"` | Explicit key indicator |
| `@decorated` | `"@decorated"` | Reserved indicator |
| `` `template` `` | `` "`template`" `` | Reserved indicator |
| `foo\rbar` *(CR embedded)* | `"foo\rbar"` | Carriage return character |
| `line1\nline2` *(newline embedded)* | `"line1\nline2"` | Embedded newline |
| `name: Martin D'vloper\nage: 26\n...` | `"name: Martin D'vloper\nage: 26\n..."` | Multiline + colon-space |

**Review by mttrbrts — APPROVED** [review]
